src/error.rs

```diff
@@ -133,25 +134,59 @@ pub trait ErrorDomain: Copy {
 
 /// Generic error used for functions that fail without any further information
 #[derive(Debug)]
-pub struct BoolError(pub &'static str);
+pub struct BoolError(Cow<'static, str>);
 
 impl BoolError {
     pub fn from_glib(b: glib_ffi::gboolean, s: &'static str) -> Result<(), Self> {
```

Comment:
By using the right trait, this should be possible to have with a single function. If you pass in a owned string, it will use that variant, otherwise the borrowed variant